src/linter/yaml/UI5YamlLinter.ts

```diff
@@ -0,0 +1,87 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import deprecatedLibraries from "../../utils/deprecatedLibs.js";
+import yaml from "js-yaml";
+import {DataWithPosition, fromYaml, getPosition} from "data-with-position";
+
+// file content schema of 'UI5Yaml' with only relevant properties
+interface UI5YamlContentSchema { // extend for further detections
+	framework: {
+		libraries: {
+			name: string;
+		}[];
+	};
+}
+
+interface UI5YamlContentSchemaWithPosInfo extends DataWithPosition {
+	framework?: {
+		libraries?: {
+			name: string;
+		}[];
+	};
+	positionKey?: {
+		end: {
+			column: number;
+			line: number;
+		};
+		start: {
+			column: number;
+			line: number;
+		};
+	};
+}
+
+export default class UI5YamlLinter {
+	#content = "";
```

Comment:
```suggestion
	#content;
```

No initial value necessary since it is set in the contructor

Reply:
e52efc0